### PR TITLE
Add sample:servicebus-queue-binder-arm

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.

### CODE REVIEW

**Comment by backwind1233** [comment]
image
@Netyyyy 
The sample runs failed, can you check this?

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    azurecaf = {
+      source  = "aztfmod/azurecaf"
+      version = "1.2.10"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "2.19.0"
+    }
+    random = {
+      source  = "hashicorp/random"
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf:16]: do we need this provider here?

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+data "azuread_client_config" "current" {}
+
+# Configure the Azure Active Directory Provider
+provider "azuread" {
+}
+
+# Configure an app
+resource "azuread_application" "servicebusqueuebinder" {
+  display_name = "servicebus-queue-binder-arm"
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf:30]: add a random string here for the display_name

like display_name = "servicebus-queue-binder-arm-xxxxx"

Comment (backwind1233) [reply servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf:30]: you can refer to https://registry.terraform.io/providers/hashicorp/random/latest/docs

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  max_delivery_count    = 10
+  lock_duration         = "PT30S"
+  max_size_in_megabytes = 1024
+  requires_session      = false
+  default_message_ttl   = "P14D"
+}
+
+resource "azurerm_role_assignment" "role_servicebus_data_contributor" {
+  scope                = azurerm_servicebus_namespace.servicebus_namespace.id
+  role_definition_name = "Contributor"
```
Comment (backwind1233) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf:102]: image

can you double-check this, it seems "Contributor" has no data plane permissions

**Review by backwind1233 — CHANGES_REQUESTED** [review]
can you run the sample again?

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
+# Spring Cloud Azure Stream Binder for Service Bus queue by ARM Authentication Sample shared library for Java
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md:1]: Spring Cloud Azure Stream Binder for Service Bus Queue by retrieving connecting string via ARM

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+
+This code sample demonstrates how to use the Spring Cloud Stream binder for
+Azure Service Bus queue by Azure Resource Manager Authentication.
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md:5]: Azure Service Bus Queue by retrieving connection string via ARM.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
+# Spring Cloud Azure Stream Binder for Service Bus queue by ARM Authentication Sample shared library for Java
+
+
+This code sample demonstrates how to use the Spring Cloud Stream binder for
+Azure Service Bus queue by Azure Resource Manager Authentication. 
+
+
+## What You Will Build
+You will build an application using Spring Cloud Stream to send and receive messages for Azure Service Bus Queue, and  a service principal for this application.
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md:9]: You will build an application using Spring Cloud Stream to send and receive messages for Azure Service Bus Queue. This sample will
- Create a Service Bus queue and a service principal and assign the `Contributor` role of the service bus to the service principal.
- Connect to the ARM using the service principal and retrieve the connection string of the service bus.
- Connect to the Service Bus queue to send and receive messages.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/pom.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  <packaging>jar</packaging>
+
+  <name>Communicate to a Single Service Bus Queue via Spring Cloud Stream By Azure Resource Manager Authentication</name>
+
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>spring-cloud-azure-resourcemanager</artifactId>
+    </dependency>
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/pom.xml:25]: we don't need this, the `spring-cloud-azure-stream-binder-servicebus` has already depended on it IIRC.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+  enable_partitioning   = false
+  max_delivery_count    = 10
+  lock_duration         = "PT30S"
+  max_size_in_megabytes = 1024
+  requires_session      = false
+  default_message_ttl   = "P14D"
+}
+
+resource "azurerm_role_assignment" "role_servicebus_data_contributor" {
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/terraform/main.tf:100]: `role_servicebus_contributor`, this role is not a data role

**Review by saragluna — CHANGES_REQUESTED** [review]

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/pom.xml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.azure.spring</groupId>
+    <artifactId>azure-spring-boot-samples</artifactId>
+    <version>1.0.0</version>
+    <relativePath>../../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spring-cloud-azure-sample-stream-binder-servicebus-queue-arm</artifactId>
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/pom.xml:14]: please also add this module into the aggregator pom, https://github.com/Azure-Samples/azure-spring-boot-samples/blob/spring-cloud-azure_4.0.0-beta.4/pom.xml.

File: servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+source ./terraform/setup_env.sh
+```
+
+#### Run with Powershell
+
+```shell
+ . terraform\setup_env.ps1
+```
+
+## Run Locally
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-stream-binder-servicebus/servicebus-queue-binder-arm/README.md:140]: Yonghui did some changes to this part, please also apply to this PR. https://github.com/Azure-Samples/azure-spring-boot-samples/pull/212

**Comment by saragluna** [comment]
> image
> 
> @Netyyyy The sample runs failed, can you check this?

@backwind1233, I can run it locally.

**Comment by backwind1233** [comment]
also, please add the new module in the root readme. @Netyyyy

**Review by saragluna — APPROVED** [review]
LGTM.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**